libguard/guard_interface.cpp

```diff
@@ -139,19 +143,31 @@ static GuardRecord getHostEndiannessRecord(const GuardRecord& record)
 void fileSizeCheck(size_t size, uint32_t count)
 {
     GuardRecord record;
+    int pos = 0;
+    int lastPos = 0;
     uint32_t aval_size = size - (count * sizeof(record) + headerSize);
 
     if (aval_size < sizeof(record))
     {
-        guard_log(GUARD_ERROR, "Size left in GUARD file is %d\n", aval_size);
-        throw GuardFileOverFlowed("Enough size is not available in GUARD file");
+        clearResolvedRecords();
```

Comment:
It will be great If you added some comments on why you are doing this - It will help for future reference.